packages/walletconnect-solana/src/adapter.ts

```diff
@@ -115,16 +117,52 @@ export class WalletConnectWallet {
         }
     }
 
-    async signTransaction(transaction: Transaction): Promise<Transaction> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
+        if (this._client && this._session) {
+            let rawTransaction: string;
+            let legacyTransaction: Transaction | undefined;
+
+            if (isVersionedTransaction(transaction)) {
+                rawTransaction = Buffer.from(transaction.serialize()).toString('base64');
+                if (transaction.version === 'legacy') {
+                    // Build Transaction for legacy sign transaction request format
+                    legacyTransaction = Transaction.from(transaction.serialize());
+                }
+            } else {
+                rawTransaction = transaction
+                    .serialize({
+                        requireAllSignatures: false,
+                        verifySignatures: false,
+                    })
+                    .toString('base64');
+                legacyTransaction = transaction;
+            }
+
+            const signature = await this._signTransaction(rawTransaction, legacyTransaction);
+            transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)));
+            return transaction;
+        } else {
+            throw new ClientNotInitializedError();
+        }
+    }
+
+    private async _signTransaction(
+        rawTransaction: string,
+        legacyTransaction?: Transaction | undefined
+    ): Promise<string> {
         if (this._client && this._session) {
             const { signature } = await this._client.request<{ signature: string }>({
                 chainId: this._network,
                 topic: this._session.topic,
-                request: { method: WalletConnectRPCMethods.signTransaction, params: { ...transaction } },
+                request: {
+                    method: WalletConnectRPCMethods.signTransaction,
+                    params: {
+                        ...legacyTransaction, // Legacy sign transaction request format
```

Comment:
@bergusman make a comment here. That it is deprecated and in the future versions only transaction will be sent.